src/azure-cli/azure/cli/command_modules/rdbms/validators.py

```diff
@@ -512,6 +512,13 @@ def _valid_range(addr_range):
     return False
 
 
+def virtual_endpoint_name_validator(ns):
+    if not re.search(r'^[A-Za-z0-9][A-Za-z0-9-]{2,62}(?<!-)$', ns.virtual_endpoint_name):
```

Comment:
this pattern matching string is not correct as this one allows capital letters to be passed for virtual endpoint name. However backend currently does not support name in caps for virtual endpoint name